### DISCUSSION TITLE
Is Trainer.validate not available now in the latest version?

### DISCUSSION THREAD

**Discussion by onpix** [Lightning-AI/pytorch-lightning#6479] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
My code is like this:

```python
model = MyLitModel()
trainer = Trainer(gpus=1)
trainer.validate(model, dataloader)
```

However, I got an `AttributeError`:

```
AttributeError: type object 'Trainer' has no attribute 'validate'
```

**Comment by carmocca** [comment, marked as answer]
It's available in master!

Note that your first call is using `Trainer.validate` not `trainer.validate` and that's why you get that error

fixed:
```python
model = MyLitModel()
trainer = Trainer(gpus=1)
trainer.validate(model, dataloader)
```

**Reply by onpix** [reply]
Sorry, my first call is a mistake when I writing the discussion. The code is actually like your fixed version. However, the errors still occur.

I try to locate `Trainer.validate` in the python console, and the output is:

<img width="845" alt="截屏2021-03-12 上午10 33 42" src="https://user-images.githubusercontent.com/32115448/110883436-6d3e7780-831e-11eb-829e-46a3869870bd.png">

But you can see `Trainer.fit` and `Trainer.test` work well. Is it a bug?

**Reply by carmocca** [reply]
Are you sure you have installed master and not from PyPI?

`pip install git+https://github.com/PyTorchLightning/pytorch-lightning`

`validate` will not be oficially released until 1.3

**Reply by onpix** [reply]
My pytorch-lightning is installed by conda and the version is 1.2.2. I'm not sure if its version is the same as master.
I reinstalled it using your command and now the problem is solved. Thanks for your help!
